block-node/messaging/src/main/java/org/hiero/block/server/messaging/impl/MessagingServiceImpl.java

```diff
@@ -0,0 +1,300 @@
+     */
+    private final Disruptor<BlockNotificationRingEvent> blockNotificationDisruptor;
+
+    /** Collect all block item handlers till start then register them with the disruptor. */
+    private final ArrayList<EventHandler<BlockItemBatchRingEvent>> blockItemHandlers = new ArrayList<>();
+
+    /** Collect all block notification handlers till start then register them with the disruptor. */
+    private final ArrayList<EventHandler<BlockNotificationRingEvent>> blockNotificationHandlers = new ArrayList<>();
+
+    /** Map of dynamic no back pressure block item handlers to their threads. So that we can stop them */
+    private final Map<NoBackPressureBlockItemHandler, Thread> dynamicNoBackPressureBlockItemHandlers = new HashMap<>();
+
+    /** Map of dynamic no back pressure block item handlers to their event processors. So that we can stop them */
+    private final Map<NoBackPressureBlockItemHandler, BatchEventProcessor<BlockItemBatchRingEvent>>
+            dynamicNoBackPressureBlockItemHandlerToEventProcessor = new HashMap<>();
+
+    /**
+     * Constructs a new MessagingServiceImpl instance with the default configuration. It uses the
+     * ConfigurationBuilderFactory to load the configuration from the classpath.
+     */
+    public MessagingServiceImpl() {
+        this(getConfig());
+    }
+
+    /**
+     * Constructs a new MessagingServiceImpl instance with the given configuration.
+     *
+     * @param config the configuration for the messaging service
+     */
+    public MessagingServiceImpl(final MessagingConfig config) {
+        blockItemDisruptor = new Disruptor<>(
+                BlockItemBatchRingEvent::new,
+                config.queueSize(),
+                VIRTUAL_THREAD_FACTORY,
+                ProducerType.SINGLE,
+                new SleepingWaitStrategy());
+        blockNotificationDisruptor = new Disruptor<>(
+                BlockNotificationRingEvent::new,
+                config.queueSize(),
+                VIRTUAL_THREAD_FACTORY,
+                ProducerType.SINGLE,
+                new SleepingWaitStrategy());
+    }
+
+    /**
+     * Get the configuration for the messaging service. It uses the ConfigurationBuilderFactory to load the
+     * configuration from the classpath. Public as it is used in tests, but is not exported from the module.
+     *
+     * @return the configuration for the messaging service
+     * @throws IllegalStateException if no ConfigurationBuilderFactory implementation is found
+     */
+    public static MessagingConfig getConfig() {
+        ConfigurationBuilderFactory configurationBuilderFactory = null;
+        final ServiceLoader<ConfigurationBuilderFactory> serviceLoader =
+                ServiceLoader.load(ConfigurationBuilderFactory.class, MessagingServiceImpl.class.getClassLoader());
+        final Iterator<ConfigurationBuilderFactory> iterator = serviceLoader.iterator();
+        if (iterator.hasNext()) {
```

Comment:
Note, this assumes we'll only ever have one implementation matching and will pick the first in the iterator if more than one.
Should we loop over and pull the expected service out based on an additional identification criteria?

Probably fine for now but something to consider down the line.